prov/shm/src/smr_progress.c

```diff
@@ -477,31 +478,42 @@ static int smr_progress_ipc(struct smr_cmd *cmd, enum fi_hmem_iface iface,
 	int64_t id;
 	int ret, fd, ipc_fd;
 	ssize_t hmem_copy_ret;
+	struct ofi_mr_entry *mr_entry;
 
 	peer_smr = smr_peer_region(ep->region, cmd->msg.hdr.id);
 	resp = smr_get_ptr(peer_smr, cmd->msg.hdr.src_data);
 
 	//TODO disable IPC if more than 1 interface is initialized
 	assert(iface == cmd->msg.data.ipc_info.iface || iface == FI_HMEM_SYSTEM);
 
+	if (cmd->msg.data.ipc_info.iface != FI_HMEM_ZE && !ep->ipc_cache) {
+		ret = ofi_ipc_cache_open(&ep->ipc_cache,
+					 ep->util_ep.domain,
+					 cmd->msg.data.ipc_info.iface);
+		if (ret)
+			goto out;
+	}
```

Comment:
Cache creation shouldn't be in a fast path.  An empty cache is just an initialized rbtree.  This should move to fabric/domain creation.